src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+
+                # TODO still valid or remove due to improved runtime of triangulation code??
+                # TODO: First compute quickly a low resolution and then
+                # compute a high resolution version.
+                # TODO: could use the size of the canvas that will display
+                # the image to save some computation time.
+
+                # Warning: allocates lot of memory, which will not be free'd until
+                # the current thread is terminated.
+
+                polar_data = angleres.AngleResolved2Polar(data_bg_corr, size, hole=False)
+
+                # TODO: don't hold too many of them in cache (eg, max 3 * 1134**2) - can this TODO be removed??
+                # TODO now we hold many more in cache
+                polar_dict[pos] = polar_data
+            except Exception:
+                logging.exception("Failed to convert to azimuthal projection")
+                return data  # display its raw as fallback
+
+        return polar_data
+
+    def _updateImage(self):
+        """Recomputes the image for the current ebeam position and polarization position requested."""
+        if not self.raw:
+            return
+
+        pos = self.stream.point.value
+        try:
+            if pos == (None, None):
+                self.image.value = None
+            else:
+                if self.stream._pos.keys()[0][-1]:
+                    pol = self.stream.polarization.value
+                else:
+                    pol = None
+                polar_data = self._project2Polar(pos + (pol,))
+
+                # update the histogram
+                # TODO: cache the histogram per image
+                # FIXME: histogram should not include the black pixels outside
+                # of the circle. => use a masked array?
+                # reset the drange to ensure that it doesn't depend on older data
+                self.stream._drange = None
+                self.stream._updateHistogram(polar_data)
+
+                self.image.value = self._project2RGB(polar_data)
+                # TODO why not store the RGB converted image in dict?? Because we need to calc the histogram
+
+        except Exception:
+            logging.exception("Updating %s image", self.__class__.__name__)
+
+    def _onPolarization(self, pos):
+        """
+        Called when the polarization VA has changed.
+        :param pos: (str) Polarization position requested.
+        """
+        self._shouldUpdateImage()
+
+    def _onBackground(self, data):  # TODO why do we need the arg data? Does not work without but dont get why?
+        """
+        Called after the background has changed.
+        :param data: (list) List of data arrays.
+        """
+        # un-cache all the polar images
+        self._polar_dict = {}
+        super(ARRawProjection, self)._onBackground(data)
+
+
+class ARPolarimetryProjection(ARProjection):
+    """
+    An ARProjection is a typical projection used to show 2D angle resolved images.
```

Comment:
Just document the difference between ARPolarimetryProjection and ARProjection/ARRawProjection.